platform/view/services/events/service.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package events
+
+import (
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view"
+	"github.com/pkg/errors"
+)
+
+type EventService interface {
+	GetPublisher() Publisher
+	GetSubscriber() Subscriber
+}
+
+type Service struct {
+	EventSystem EventSystem
+}
+
+func (s *Service) GetSubscriber() Subscriber {
+	return s.EventSystem
+}
+
+func (s *Service) GetPublisher() Publisher {
+	return s.EventSystem
+}
+
+func getService(sp view.ServiceProvider) (EventService, error) {
+	s, err := sp.GetService(&Service{})
```

Comment:
what about making &Service{} a global var that is initiazialed once and for all? This will avoid instantiating an empty object each time a getService is invoked.

Reply:
Pushed a fixup addressing your comment! Thanks Angelo